src/ConfigSetup.cpp

```diff
@@ -805,28 +820,43 @@ void ConfigSetup::Init(const char *fileName, MultiSim const*const& multisim)
       printf("%-40s %-4.4f \n", "Info: Molecule swap move frequency",
              sys.moves.transfer);
     } else if(CheckString(line[0], "MEMC-1Freq")) {
-      sys.moves.memc = stringtod(line[1]);
-      printf("%-40s %-4.4f \n", "Info: MEMC-1 move frequency",
+      if(stringtod(line[1]) > 0.0){
+        if(sys.memcVal.enable){
+          std::cout << "Error: Only one type of MEMC move is supported at a time." << std::endl;
+          exit(EXIT_FAILURE);
+        } else {
+          sys.moves.memc = stringtod(line[1]);
+          printf("%-40s %-4.4f \n", "Info: MEMC-1 move frequency",
              sys.moves.memc);
-      if(sys.moves.memc > 0.0) {
-        sys.memcVal.enable = true;
-        sys.memcVal.MEMC1 = true;
+          sys.memcVal.enable = true;
+          sys.memcVal.MEMC1 = true;
+        }
       }
     } else if(CheckString(line[0], "MEMC-2Freq")) {
-      sys.moves.memc = stringtod(line[1]);
-      printf("%-40s %-4.4f \n", "Info: MEMC-2 move frequency",
+      if(stringtod(line[1]) > 0.0){
+        if(sys.memcVal.enable){
+          std::cout << "Error: Only one type of MEMC move is supported at a time." << std::endl;
+          exit(EXIT_FAILURE);
+        } else {
+          sys.moves.memc = stringtod(line[1]);
+          printf("%-40s %-4.4f \n", "Info: MEMC-2 move frequency",
              sys.moves.memc);
-      if(sys.moves.memc > 0.0) {
-        sys.memcVal.enable = true;
-        sys.memcVal.MEMC2 = true;
+          sys.memcVal.enable = true;
+          sys.memcVal.MEMC2 = true;
+        }
       }
     } else if(CheckString(line[0], "MEMC-3Freq")) {
-      sys.moves.memc = stringtod(line[1]);
-      printf("%-40s %-4.4f \n", "Info: MEMC-3 move frequency",
+      if(stringtod(line[1]) > 0.0){
+        if(sys.memcVal.enable){
+          std::cout << "Error: Only one type of MEMC move is supported at a time." << std::endl;
+          exit(EXIT_FAILURE);
```

Comment:
This is not the way I asked. We should not do error checking here. It's already done in verifyInputs() function. Just avoid overriding the value.